apps/api/src/models/post.ts

```diff
@@ -0,0 +1,45 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+interface ICategory {
+  name: string;
+}
+
+interface IComment {
+  author: string;
+  content: string;
+}
+interface IPost extends Document {
+  title: string;
+  image?: string;
+  description: string;
+  category: ICategory;
```

Comment:
In this case since `IPost` is only being use by the `Schema` and the type of category is `ObjectId` this can also be and `ObjectId`

```suggestion
  category: mongoose.Types.ObjectId;
```